packages/wallets/src/features/cashier/modules/Transactions/components/TransactionsCompletedRow/TransactionsCompletedRow.tsx

```diff
@@ -12,11 +12,37 @@ type TProps = {
 };
 
 const TransactionsCompletedRow: React.FC<TProps> = ({ accounts, transaction, wallet }) => {
+    // TODO: remove this once backend adds `to` and `from` for Deriv X transfers
+    const dxtradeToFrom = useMemo(() => {
+        if (
+            transaction?.action_type !== 'transfer' ||
+            !transaction.longcode ||
+            !transaction.longcode.includes('Deriv X')
+        )
+            return null;
+        const longcodeMessageTokens = transaction.longcode.split(' ');
+        const direction = longcodeMessageTokens[1];
+        const dxtradeLoginid = longcodeMessageTokens.find(token => token.startsWith('DX'));
+        return {
+            from: { loginid: wallet.loginid },
+            to: { loginid: wallet.loginid },
+            ...(direction && { [direction]: { loginid: dxtradeLoginid } }),
+        };
```

Comment:
The `longcode` for a Deriv X transfer is a string and looks like this: `` `Transfer ${'to' | 'from'} Deriv X account 
 ${loginid}` ``. We split it into tokens/words and use those to get the transaction details

Reply:
Nice trick to replace the key 😄.